tests/projects/c++/modules/circular_dependency/test.lua

```diff
@@ -0,0 +1,54 @@
+import("lib.detect.find_tool")
+import("core.base.semver")
+
+function _build()
+    local ci = (os.getenv("CI") or os.getenv("GITHUB_ACTIONS") or ""):lower()
+    local passed = false
+    try {function()
+        if ci == "true" then
+            os.run("xmake -rvD")
+        else
+            os.run("xmake -r")
+        end
+    end, catch{function(errors)
+        print(errors)
+        passed = errors:match("circular modules dependency detected!")
+    end}}
+
+    assert(passed)
+end
+
+function main(t)
```

Comment:
```lua
import(".test_base", {alias = "test_build"})

function main(t)
    t:will_raise(function() test_build() end, "circular modules dependency detected!") 
end
```

Reply:
i tried this but doesn't work

```
"F:\repos\xmake\tests\projects\c++\modules\circular_dependency\test.lua"
>> testing tests\projects\c++\modules\circular_dependency ...
>>     running test_main ...
checking for clang ... ✔
checking for platform ... windows
checking for architecture ... x64
[  0%]: <circular_dependency> generating.module.deps src\main.cpp
[  0%]: <circular_dependency> generating.module.deps src\hello.mpp
[  0%]: <circular_dependency> generating.module.deps src\hello2.mpp
[  0%]: <circular_dependency> generating.module.deps src\hello3.mpp
❗ error: circular modules dependency detected!
hello
   -> import hello3
   -> import hello2
   -> import hello
>>     test failed: expected raise with message *error: circular modules dependency detected!* but got .\xmake\core\sandbox\modules\os.lua:337: exec(xmake -r) failed(-1)
stack traceback:
    [C]: in function 'error'
    [.\xmake\core\base\os.lua:957]:
    [.\xmake\core\sandbox\modules\os.lua:337]: in function 'exec'
    [...rojects\c++\modules\circular_dependency\..\test_base.lua:9]: in function '_build'
    [...rojects\c++\modules\circular_dependency\..\test_base.lua:18]:

>>       function test_main tests\projects\c++\modules\circular_dependency\test.lua:3
❗ error: .\xmake\core\main.lua:306: .\tests\runner.lua:57: .\tests\test_utils\print_error.lua:46: aborting because of failed assertion ...
stack traceback:
    [C]: in function 'error'
    [.\xmake\core\base\os.lua:957]:
    [.\tests\test_utils\print_error.lua:46]: in function 'print_error'
    [.\tests\test_utils\test_assert.lua:61]: in function 'finally'
    [.\xmake\core\sandbox\modules\try.lua:129]: in function 'try'
    [.\tests\test_utils\test_assert.lua:51]: in function 'will_raise'
    [.\tests\projects\c++\modules\circular_dependency\test.lua:4]:

stack traceback:
        [C]: in function 'error'
        .\xmake\core\base\os.lua:957: in function 'os.raiselevel'
        (...tail calls...)
        .\xmake\core\main.lua:306: in upvalue 'cotask'
        .\xmake\core\base\scheduler.lua:404: in function <.\xmake\core\base\scheduler.lua:397>
❗ error: .\xmake\core\main.lua:306: .\xmake\core\sandbox\modules\os.lua:378: execv(xmake lua -v -D tests\runner.lua F:\repos\xmake\tests\projects\c++\modules\circular_dependency\test.lua) failed(-1)
stack traceback:
    [C]: in function 'error'
    [.\xmake\core\base\os.lua:957]:
    [.\xmake\core\sandbox\modules\os.lua:378]: in function 'execv'
    [.\tests\run.lua:14]: in function '_run_test'
    [.\tests\run.lua:52]:
    [.\xmake\plugins\lua\main.lua:124]:
    [C]: in function 'xpcall'
    [.\xmake\core\base\utils.lua:280]:
    [.\xmake\core\base\task.lua:501]: in function 'run'
    [.\xmake\core\main.lua:304]: in function 'cotask'
    [.\xmake\core\base\scheduler.lua:404]:

stack traceback:
        [C]: in function 'error'
        .\xmake\core\base\os.lua:957: in function 'base/os.raiselevel'
        (...tail calls...)
        .\xmake\core\main.lua:306: in upvalue 'cotask'
        .\xmake\core\base\scheduler.lua:404: in function <.\xmake\core\base\scheduler.lua:397>
 ```

Reply:
we need use `os.run("xmake -r")` instead of os.exec in test_base/_build